File: demo-android/app/src/main/java/com/example/demo/repository/TaskRepository.kt
package com.example.demo.repository

import com.example.demo.data.model.Task
import com.example.demo.data.model.TaskRequest
import com.example.demo.data.remote.TaskApi

class TaskRepository(
    private val api: TaskApi
) {
    suspend fun getTasks(): List<Task> = api.getTasks()

    suspend fun getTask(taskId: String): Task = api.getTask(taskId)

    suspend fun isValid(taskId: String): Boolean = api.isValid(taskId)

    suspend fun createTask(request: TaskRequest): Task = api.createTask(request)

    suspend fun updateTask(taskId: String, request: TaskRequest): Task =
        api.updateTask(taskId, request)

    suspend fun deleteTask(taskId: String) {
        api.deleteTask(taskId)
    }

}

File: demo-react-native/src/repository/taskRepository.ts
import {Task, TaskRequest} from '@/data/models/task';
import {taskApi} from '@/data/remote/taskApi';

export const taskRepository = {
  getTasks: (): Promise<Task[]> => taskApi.getTasks(),

  getTask: (taskId: string): Promise<Task> => taskApi.getTask(taskId),

  isValid: (taskId: string): Promise<boolean> => taskApi.isValid(taskId),

  createTask: (request: TaskRequest): Promise<Task> =>
    taskApi.createTask(request),

  updateTask: (taskId: string, request: TaskRequest): Promise<Task> =>
    taskApi.updateTask(taskId, request),

  deleteTask: (taskId: string): Promise<void> => taskApi.deleteTask(taskId),
};
